src/Idris/Package/Types.idr

```diff
@@ -85,6 +98,39 @@ inBounds mv bounds
                            then v <= v'
                            else v < v') bounds.upperBound
 
+namespace Version
+  ||| Check if a Version is within the bounds of a PkgVersionBounds.
+  |||
+  ||| In addition to comparing major, minor, and patch version numbers,
+  ||| a Version with a tag is always considered larger than one with the
+  ||| same major, minor, and patch numbers but no tag.
+  |||
+  ||| Therefore, Version 0.1.0-abcd will be within the bounds >0.1.0 && <0.2.0.
+  ||| Similarly, Version 0.2.0-abcd will be _outside_ the bounds
+  ||| >0.1.0 && <=0.2.0.
+  export
+  inBounds : Version -> PkgVersionBounds -> Bool
+  inBounds (MkVersion (maj, min, patch) tag) bounds
+     = let v = MkPkgVersion (maj ::: [min, patch]) in
+       maybe True (\v' => if bounds.lowerInclusive
+                             then v >= v'
+                             else v > v' || (v == v' && tag /= Nothing)) bounds.lowerBound &&
+       maybe True (\v' => if bounds.upperInclusive
+                             then v < v' || (v == v' && tag == Nothing)
+                             else v < v' || (v == v' && tag /= Nothing)) bounds.upperBound
```

Comment:
```suggestion
                             else v < v') bounds.upperBound
```

otherwise `2.0-abc < 2.0` which seems to go against the logic that 

> a Version with a tag is always considered larger than one with the
> same major, minor, and patch numbers but no tag.